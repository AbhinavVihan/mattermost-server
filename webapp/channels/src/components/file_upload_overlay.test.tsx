// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import FileUploadOverlay from 'components/file_upload_overlay';
import {renderWithIntl} from 'tests/react_testing_utils';

describe('components/FileUploadOverlay', () => {
    test('should match snapshot when file upload is showing with no overlay type', () => {
        const {container} = renderWithIntl(
            <FileUploadOverlay
                overlayType=''
            />,
        );

        container.className.includes('file-overlay');

        expect(container).toMatchSnapshot();
    });

    test('should match snapshot when file upload is showing with overlay type of right', () => {
        const {container} = renderWithIntl(
            <FileUploadOverlay
                overlayType='right'
            />,
        );

        container.className.includes('right-file-overlay');
        expect(container).toMatchSnapshot();
    });

    test('should match snapshot when file upload is showing with overlay type of center', () => {
        const {container} = renderWithIntl(
            <FileUploadOverlay
                overlayType='center'
            />,
        );
        container.className.includes('center-file-overlay');

        expect(container).toMatchSnapshot();
    });
});
