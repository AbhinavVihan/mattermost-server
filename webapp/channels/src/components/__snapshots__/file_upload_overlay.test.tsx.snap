// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FileUploadOverlay should match snapshot when file upload is showing with no overlay type 1`] = `
<div>
  <div
    class="file-overlay hidden"
  >
    <div
      class="overlay__indent"
    >
      <div
        class="overlay__circle"
      >
        <img
          alt="Files"
          class="overlay__files"
          src="fileOverlayImage"
        />
        <span>
          <i
            class="fa fa-upload"
            title="Upload Icon"
          />
          Drop a file to upload it.
        </span>
        <img
          alt="Logo"
          class="overlay__logo"
          src="overlayLogoImage"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`components/FileUploadOverlay should match snapshot when file upload is showing with overlay type of center 1`] = `
<div>
  <div
    class="file-overlay hidden center-file-overlay"
  >
    <div
      class="overlay__indent"
    >
      <div
        class="overlay__circle"
      >
        <img
          alt="Files"
          class="overlay__files"
          src="fileOverlayImage"
        />
        <span>
          <i
            class="fa fa-upload"
            title="Upload Icon"
          />
          Drop a file to upload it.
        </span>
        <img
          alt="Logo"
          class="overlay__logo"
          src="overlayLogoImage"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`components/FileUploadOverlay should match snapshot when file upload is showing with overlay type of right 1`] = `
<div>
  <div
    class="file-overlay hidden right-file-overlay"
  >
    <div
      class="overlay__indent"
    >
      <div
        class="overlay__circle"
      >
        <img
          alt="Files"
          class="overlay__files"
          src="fileOverlayImage"
        />
        <span>
          <i
            class="fa fa-upload"
            title="Upload Icon"
          />
          Drop a file to upload it.
        </span>
        <img
          alt="Logo"
          class="overlay__logo"
          src="overlayLogoImage"
        />
      </div>
    </div>
  </div>
</div>
`;
